backend/src/test/java/com/hackner/musicompass/controller/RecommendationControllerTest.java

```diff
@@ -0,0 +1,122 @@
+package com.hackner.musicompass.controller;
+
+import com.hackner.musicompass.controller.model.RecommendationTagsDto;
+import com.hackner.musicompass.db.ArtistMongoDb;
+import com.hackner.musicompass.db.RecommendationCategoryMongoDb;
+import com.hackner.musicompass.model.Artist;
+import com.hackner.musicompass.model.RecommendationCategory;
+import com.hackner.musicompass.model.RecommendationTags;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+
+import java.lang.reflect.Type;
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class RecommendationControllerTest {
+
+    @LocalServerPort
+    private int port;
+
+    private String getUrl() {return "http://localhost:" + port + "api/recommendation";}
+
+    @Autowired
+    private TestRestTemplate testRestTemplate;
+
+    @Autowired
+    private ArtistMongoDb artistMongoDb;
+
+    @Autowired
+    private RecommendationCategoryMongoDb recommendationCategoryMongoDb;
+
+    @BeforeEach
+    public void setup(){artistMongoDb.deleteAll();}
+
+    @Test
+    @DisplayName("Get recommendation category values")
+    public void getRecommendationCategoryValues(){
+        //GIVEN
+        recommendationCategoryMongoDb.save(RecommendationCategory.builder().categoryName("Genre").categoryValues(Arrays.asList("Genre1", "Genre2")).build());
+        recommendationCategoryMongoDb.save(RecommendationCategory.builder().categoryName("Role").categoryValues(Arrays.asList("Role1","Role2")).build());
```

Comment:
When you want Lists as values I suggest using List.of() instead of Arrays.asList(), so you save yourself the type parse from array to list.

Reply:
the syntax seemed like a workaround to me before but it is working well. Thx for the input - I will use this from now on